### ISSUE TITLE
'NoneType' object has no attribute 'neighbors'

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#517]
In GitLab by @mcastril on Apr 8, 2020, 17:28

I report this issue that is the first time that I see happening, I thing.

I create an experiment with 3.12.0b, I move to 3.12.1b and try to run it. But the error prompts:

```
RITICAL] Unhandled exception on Autosubmit: 'NoneType' object has no attribute 'neighbors'
Traceback (most recent call last):
  File "/shared/earth/software/autosubmit/3.12.1b-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.1b0-py2.7.egg/autosubmit/autosubmit.py", line 440, in parse_args
    return Autosubmit.run_experiment(args.expid, args.notransitive,args.update_version)
  File "/shared/earth/software/autosubmit/3.12.1b-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.1b0-py2.7.egg/autosubmit/autosubmit.py", line 1104, in run_experiment
    job_list = Autosubmit.load_job_list(expid, as_conf, notransitive=notransitive)
  File "/shared/earth/software/autosubmit/3.12.1b-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.1b0-py2.7.egg/autosubmit/autosubmit.py", line 3826, in load_job_list
    new=False, notransitive=notransitive)
  File "/shared/earth/software/autosubmit/3.12.1b-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.1b0-py2.7.egg/autosubmit/job/job_list.py", line 155, in generate
    self.update_genealogy(new, notransitive)
  File "/shared/earth/software/autosubmit/3.12.1b-foss-2015a-Python-2.7.9/lib/python2.7/site-packages/autosubmit-3.12.1b0-py2.7.egg/autosubmit/job/job_list.py", line 1247, in update_genealogy
    child for child in job.children if child.name not in self.graph.neighbors(job.name)]
AttributeError: 'NoneType' object has no attribute 'neighbors'
```

Any idea @wuruchi @dbeltrankyl ?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 8, 2020, 17:31

Hi,

I updated 3.12.1 with job_list @wuruchi  changes so maybe there is a retrocompatability issue?

**Comment by kinow** [comment]
In GitLab by @mcastril on Apr 8, 2020, 17:33

That's my first guess, something new not created by 3.12.0b.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 17:34

Hello @mcastril 

I suppose 3.12.0b did not create the structure database that 3.12.1b is trying to call. I will add more validation. What is the experiment id you are working with?

**Comment by kinow** [comment]
In GitLab by @mcastril on Apr 8, 2020, 17:35

a2rj, I thought it was in the logs, sorry

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 17:48

mentioned in commit 6c8f681b43848c780c51c53613b795b64b864220

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 17:50

@mcastril 

I have opened a merge request !162 that should solve it.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 17:52

mentioned in commit 2ad8f997cd2f6e51e029465e2816520fe4a3912d

**Comment by kinow** [comment]
In GitLab by @mcastril on Apr 8, 2020, 18:00

Thanks team for your promptness. I will run a test.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 18:03

@mcastril
I think I changed my user by mistake in a2rj/conf/platforms.conf

**Comment by kinow** [comment]
In GitLab by @mcastril on Apr 8, 2020, 18:06

It is working now.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 18:08

Great. I will run a couple more tests and see how it develops.

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 18:31

mentioned in commit b2166aeba04b08241aba365e67dc9aa6ac54d6b4

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 18:33

mentioned in commit 1c1884098c64ead102aa39ed39aba9941b64fa89

**Comment by kinow** [comment]
In GitLab by @wuruchi on Apr 8, 2020, 18:34

Merged !163 in an attempt to cover any other use cases.

**Comment by kinow** [comment]
In GitLab by @jeronimoescribano on Apr 8, 2020, 21:37

Thank you very much @wuruchi , @dbeltrankyl and @mcastril  .  I've tried now and it works for me. 

The only issue, that maybe is unrelated to this one, is that now the "autosubmit recovery" is taking a lot of time. In fact, this command in 3.12.1 was really fast before, but now it has the same speed of the 3.12.0. Should I open a new issue?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Apr 8, 2020, 22:55

@jeronimoescribano ,

Sorry, it is fixed but not updated, forgot to add it to 3.12.1b. Tomorrow morning will be solved in both versions
